src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2247,6 +2247,42 @@ internal void AddModuleInitializerMethod(MethodSymbol method)
             LazyInitializer.EnsureInitialized(ref _moduleInitializerMethods).Add(method);
         }
 
+        private ConcurrentSet<(InterceptsLocationAttributeData, MethodSymbol)>? _interceptions;
+
+        internal void AddInterception(InterceptsLocationAttributeData location, MethodSymbol interceptor)
+        {
+            Debug.Assert(!_declarationDiagnosticsFrozen);
+            LazyInitializer.EnsureInitialized(ref _interceptions).Add((location, interceptor));
+        }
+
+        internal (InterceptsLocationAttributeData data, MethodSymbol interceptor)? GetInterceptor(Location? callLocation)
+        {
+            if (_interceptions is null || callLocation is null)
+            {
+                return null;
+            }
+
+            foreach (var (interceptsLocation, interceptor) in _interceptions)
+            {
+                var callLineColumn = callLocation.GetLineSpan().Span.Start;
```

Comment:
Consider extracting the parts of `callLocation` before the loop starts, since they don't vary.